src/Microsoft.AspNet.Hosting/Logging/RequestInformationValues.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Framework.Logging;
+
+namespace Microsoft.AspNet.Hosting
+{
+    public class RequestInformationValues : ILogValues
+    {
+        public RequestInformationValues(Guid requestId)
+        {
+            RequestId = requestId;
+        }
+
+        public Guid RequestId { get; }
+
+        /// <inheritdoc />
+        public string Format()
+        {
+            return LogFormatter.FormatLogValues(this);
+        }
+
+        /// <inheritdoc />
+        public IEnumerable<KeyValuePair<string, object>> GetValues()
+        {
+            return new[] 
```

Comment:
First, it's tons clearer that we don't want an array, we just want one ```KeyValuePair<string, object>```

Second, unless someone calls ```.Next()``` and ```.Current```, the actual ```KeyValuePair<string, object>``` is never allocated.

Third, this is an allocation a ```KeyValuePair<string, object>```, then an allocation of an array and a ```.Add()``` on that array. This array is then returned, which on call to ```IEnumerable<KeyValuePair<string, object>>GetEnumerator()``` returns an ```IEnumerator<KeyValuePair<string, object>>```. That enumerator's code is quite small, but still exists as an allocation.

Meanwhile, the ```yield return``` version news up a mini-state machine (about a wash for the new of the array) which directly handles the ```IEnumerator<KeyValuePair<string, object>>``` logic (thus no second allocation).